CMakeLists.txt

```
@@ -526,6 +536,29 @@ if(FLB_METRICS)
   FLB_DEFINITION(FLB_HAVE_METRICS)
 endif()
 
+# WASM
+if(FLB_WASM)
+  if (FLB_SYSTEM_LINUX)
+    check_c_source_compiles("
+       #include <stdatomic.h>
+       int main() {
+          _Atomic int a;
+          return 0;
+       }" FLB_HAVE_STDATOMIC_H)
+    check_include_files(stdatomic.h FLB_HAVE_STDATOMIC_H)
+    if (FLB_HAVE_STDATOMIC_H)
+      enable_language (ASM)
+      FLB_DEFINITION(FLB_HAVE_WASM)
+    else ()
+      message(STATUS "This platform is too old to support FLB_WASM feature. Now disabled.")
```

`too old` is probably not right, maybe just have 
```suggestion
      message(STATUS "This platform does not support the FLB_WASM feature so disabled.")
```

Implicit modification of flags though is probably not right: if someone explicitly enables FLB_WASM then I'd expect a build failure if I have missed dependencies or other set up.

---

This mainly aimed for CentOS 7 and perhaps CentOS 6.
Enabling WASM feature is harmful to build for these platforms.
For removing this operation, we have to install devtoolset that provides stdatomic.h.

---

Fixed in https://github.com/fluent/fluent-bit/pull/5931/commits/63b9a285e761da6e41a77e815826344dc95252aa.